modules/avatar/react/lib/Avatar.tsx

```diff
@@ -10,80 +11,136 @@ export enum AvatarVariant {
-  borderRadius: borderRadius.circle,
-  boxSizing: 'border-box',
-  overflow: 'hidden',
-  '& img': {
-    width: '100%',
-    height: '100%',
+const StyledContainer = styled('button', {
+  shouldForwardProp: prop => isPropValid(prop) && prop !== 'size',
+})<Pick<AvatarProps, 'variant' | 'size' | 'onClick'>>(
+  {
+    background: colors.soap200,
+    position: 'relative',
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    padding: 0,
+    border: 0,
+    boxSizing: 'border-box',
+    overflow: 'hidden',
+    borderRadius: borderRadius.circle,
   },
-};
+  ({size}) => ({
+    height: size,
+    width: size,
+  }),
+  ({variant, onClick}) => ({
+    cursor: onClick ? 'pointer' : 'default',
+    '&:not([disabled])': {
+      '&:focus': {
+        outline: 'none',
+        ...(variant === AvatarVariant.Dark ? focusRing(2, 2) : focusRing(2)),
+      },
+    },
+    ...hideMouseFocus,
+  })
+);
 
-const StyledContainer = styled('div', {
-  shouldForwardProp: prop => isPropValid(prop) && prop !== 'size',
-})<Pick<AvatarProps, 'variant' | 'size'>>(
+const StyledStack = styled('span')<Pick<AvatarProps, 'size'>>(
   {
-    ...avatarStyles,
+    position: 'absolute',
   },
   ({size}) => ({
     height: size,
     width: size,
   })
 );
 
-export default class Avatar extends React.Component<AvatarProps> {
-  static Variant = AvatarVariant;
-  static Size = SystemIconCircleSize;
+const StyledImage = styled('img')<{isLoaded: boolean}>(
+  {
+    width: '100%',
+    height: '100%',
+    borderRadius: borderRadius.circle,
+    transition: 'opacity 200ms linear',
+  },
+  ({isLoaded}) => ({
+    opacity: isLoaded ? 1 : 0,
+  })
+);
+
+const Avatar = ({
+  variant = AvatarVariant.Light,
+  size = SystemIconCircleSize.m,
+  altText = 'Avatar',
+  url,
+  onClick,
+  buttonRef,
+  ...elemProps
+}: AvatarProps) => {
+  const [imageLoaded, setImageLoaded] = useState(false);
 
-  static defaultProps = {
-    variant: AvatarVariant.Light,
-    size: SystemIconCircleSize.m,
-    altText: 'Avatar',
+  const loadImage = () => {
+    if (!imageLoaded) {
+      setImageLoaded(true);
+    }
   };
 
-  render() {
-    const {variant, altText, size, url, ...elemProps} = this.props;
+  const background = variant === AvatarVariant.Dark ? colors.blueberry400 : colors.soap300;
 
-    const background = variant === AvatarVariant.Dark ? colors.blueberry400 : colors.soap300;
-    return (
-      <StyledContainer variant={variant} size={size} aria-label={altText} {...elemProps}>
-        {url ? (
-          <img src={url} alt={altText} />
-        ) : (
-          <SystemIconCircle icon={userIcon} background={background} size={size} />
-        )}
-      </StyledContainer>
-    );
-  }
-}
+  return (
+    <StyledContainer
+      variant={variant}
+      size={size}
+      aria-label={altText}
+      onClick={onClick}
+      disabled={onClick ? false : true}
+      ref={buttonRef}
+      {...elemProps}
+    >
+      <StyledStack size={size}>
```

Comment:
Where does the term `Stack` come from?

Reply:
🤷‍♂️ it is used to stack elements on top of each other using absolute position. Have I ever mentioned how much I hate naming styled components.

Reply:
Ah - makes sense. Haha I feel you 😂